File: ChatHub.cs
using Microsoft.AspNetCore.SignalR;

namespace Chat_SignalIR
{
    public class ChatHub:Hub
    {
    
        public async Task SendMessage(string user, string text)
        {
            var msg = new Message
            {
                Timestamp = DateTime.Now,
                User = user,
                Text = text
            }; 
            await Clients.All.SendAsync("ReceiveMessage", msg);
        }
    }
}

File: wwwroot/js/ChatHub.js
"use strict";
var connection = new
    signalR.HubConnectionBuilder().withUrl("/chatHub").build();

connection.on("ReceiveMessage", function (msg) {
    console.log("ReceiveMessage")
    console.log(msg)
    var li= document.createElement("li");
    li.textContent = msg.user + ": " + msg.text+":"+msg.timestamp;
    document.getElementById("messagesList").appendChild(li);


});

connection.start().then(function () {
    console.log("SignalR started!");

    document.getElementById("someButton").addEventListener("click", function () {
        var user = document.getElementById("someId").value;
        var message = document.getElementById("someOtherId").value;

        connection.invoke("SendMessage", user, message).catch(function (err) {
            return console.error(err.toString());
        });
        document.getElementById("someOtherId").value = "";
    });

}).catch(function (err) {
    return console.error(err.toString());
});
